src/util/piped_process.cpp

```diff
@@ -0,0 +1,388 @@
+    FILE_SHARE_READ | FILE_SHARE_WRITE,             // Shared read/write
+    &sec_attr,                                      // Need this for inherit
+    OPEN_EXISTING,                                  // Opening other end
+    FILE_ATTRIBUTE_NORMAL | FILE_FLAG_NO_BUFFERING, // Normal, but don't buffer
+    NULL);
+  if(child_std_IN_Wr == INVALID_HANDLE_VALUE)
+  {
+    throw std::runtime_error("Input pipe creation failed for child_std_IN_Wr");
+  }
+  if(!SetHandleInformation(child_std_IN_Rd, HANDLE_FLAG_INHERIT, 0))
+  {
+    throw std::runtime_error(
+      "Input pipe creation failed on SetHandleInformation");
+  }
+  tmp_name = base_name;
+  tmp_name.append("\\OUT");
+  tmp_str = tmp_name.c_str();
+  child_std_OUT_Rd = CreateNamedPipe(
+    tmp_str,
+    PIPE_ACCESS_INBOUND,          // Reading for us
+    PIPE_TYPE_BYTE | PIPE_NOWAIT, // Bytes and non-blocking
+    PIPE_UNLIMITED_INSTANCES,     // Probably doesn't matter
+    BUFSIZE,
+    BUFSIZE,    // Output and input bufffer sizes
+    0,          // Timeout in ms, 0 = use system default
+    &sec_attr); // For inheritance by child
+  if(child_std_OUT_Rd == INVALID_HANDLE_VALUE)
+  {
+    throw std::runtime_error(
+      "Output pipe creation failed for child_std_OUT_Rd");
+  }
+  child_std_OUT_Wr = CreateFileA(
+    tmp_str,
+    GENERIC_WRITE,                                  // Write side
+    FILE_SHARE_READ | FILE_SHARE_WRITE,             // Shared read/write
+    &sec_attr,                                      // Need this for inherit
+    OPEN_EXISTING,                                  // Opening other end
+    FILE_ATTRIBUTE_NORMAL | FILE_FLAG_NO_BUFFERING, // Normal, but don't buffer
+    NULL);
+  if(child_std_OUT_Wr == INVALID_HANDLE_VALUE)
+  {
+    throw std::runtime_error(
+      "Output pipe creation failed for child_std_OUT_Wr");
+  }
+  if(!SetHandleInformation(child_std_OUT_Rd, HANDLE_FLAG_INHERIT, 0))
+  {
+    throw std::runtime_error(
+      "Output pipe creation failed on SetHandleInformation");
+  }
+  // Create the child process
+  STARTUPINFOW start_info;
+  BOOL success = FALSE;
+  ZeroMemory(&proc_info, sizeof(PROCESS_INFORMATION));
+  ZeroMemory(&start_info, sizeof(STARTUPINFOW));
+  start_info.cb = sizeof(STARTUPINFOW);
+  start_info.hStdError = child_std_OUT_Wr;
+  start_info.hStdOutput = child_std_OUT_Wr;
+  start_info.hStdInput = child_std_IN_Rd;
+  start_info.dwFlags |= STARTF_USESTDHANDLES;
+  // Unpack the command into a single string for Windows API
+  std::wstring cmdline = widen(commandvec[0]);
+  for(int i = 1; i < commandvec.size(); i++)
+  {
+    cmdline.append(L" ");
+    cmdline.append(quote_windows_arg(widen(commandvec[i])));
+  }
+  // Note that we do NOT free this since it becomes part of the child
+  // and causes heap corruption in Windows if we free!
+  WCHAR *szCmdline =
+    reinterpret_cast<WCHAR *>(malloc(sizeof(WCHAR) * cmdline.size()));
+  wcscpy(szCmdline, cmdline.c_str());
```

Comment:
You need to add one `WCHAR` to the size of the allocated buffer for the terminating zero.
To avoid this problem altogether, use `_wcsdup`.

Reply:
Done